src/o365/spo/commands/search/search.ts

```diff
@@ -0,0 +1,434 @@
+    const sortListRequestString = args.options.sortList ? `&sortList='${encodeURIComponent(args.options.sortList)}'` : ``;
+    const rankingModelIdRequestString = args.options.rankingModelId ? `&rankingmodelid='${args.options.rankingModelId}'` : ``;
+    const propertiesRequestString = this.getPropertiesRequestString(args);
+    const refinersRequestString = args.options.refiners ? `&refiners='${args.options.refiners}'` : ``;
+    const hiddenConstraintsRequestString = args.options.hiddenConstraints ? `&hiddenconstraints='${args.options.hiddenConstraints}'` : ``;
+    const clientTypeRequestString = args.options.clientType ? `&clienttype='${args.options.clientType}'` : ``;
+    const enablePhoneticRequestString = typeof(args.options.enablePhonetic) === 'undefined' ? `` : `&enablephonetic=${args.options.enablePhonetic}`;    
+    const processBestBetsRequestString = typeof(args.options.processBestBets) === 'undefined' ? `` : `&processbestbets=${args.options.processBestBets}`;
+    const enableQueryRulesRequestString = typeof(args.options.enableQueryRules) === 'undefined' ? `` : `&enablequeryrules=${args.options.enableQueryRules}`;
+    const processPersonalFavoritesRequestString = typeof(args.options.processPersonalFavorites) === 'undefined' ? `` : `&processpersonalfavorites=${args.options.processPersonalFavorites}`;
+
+    //Construct single requestUrl
+    const requestUrl = `${webUrl}/_api/search/query?querytext='${args.options.query}'`.concat(
+      propertySelectRequestString,
+      startRowRequestString,
+      rowLimitRequestString,
+      sourceIdRequestString,
+      trimDuplicatesRequestString,
+      enableStemmingRequestString,
+      cultureRequestString,
+      refinementFiltersRequestString,
+      queryTemplateRequestString,
+      sortListRequestString,
+      rankingModelIdRequestString,
+      propertiesRequestString,
+      refinersRequestString,
+      hiddenConstraintsRequestString,
+      clientTypeRequestString,
+      enablePhoneticRequestString,
+      processBestBetsRequestString,
+      enableQueryRulesRequestString,
+      processPersonalFavoritesRequestString
+    );
+
+    if(this.debug) {
+      cmd.log(`RequestURL: ${requestUrl}`);
+    }
+    return requestUrl;
+  }
+
+  private getPropertiesRequestString(args: CommandArgs):string {
+    let properties = args.options.properties ? args.options.properties : '';
+    if(args.options.sourceName) {
+      if(properties && !properties.endsWith(",")) { properties += `,`; }
+      properties += `SourceName:${args.options.sourceName},SourceLevel:SPSite`;
+    }
+    return properties ? `&properties='${properties}'` : ``;
+  }
+
+  private getSelectPropertiesArray(args: CommandArgs) {
+    return args.options.selectProperties 
+      ? args.options.selectProperties.split(",")
+      : (!args.options.output || args.options.output === "text") 
+        ? ["Rank","DocId","OriginalPath"/*,"PartitionId","UrlZone","Culture","ResultTypeId","RenderTemplateId"*/]
```

Comment:
For the default output in text mode, let's use Title and OriginalPath properties. Rank and DocId don't say much